test/Microsoft.AspNetCore.Mvc.Razor.Host.Test/ViewComponentTagHelperPassTest.cs

```diff
@@ -0,0 +1,337 @@
+{
+    private readonly global::Microsoft.AspNetCore.Mvc.IViewComponentHelper _helper = null;
+    public __Generated__TagCloudViewComponentTagHelper(global::Microsoft.AspNetCore.Mvc.IViewComponentHelper helper)
+    {
+        _helper = helper;
+    }
+    [Microsoft.AspNetCore.Razor.TagHelpers.HtmlAttributeNotBoundAttribute, global::Microsoft.AspNetCore.Mvc.ViewFeatures.ViewContextAttribute]
+    public global::Microsoft.AspNetCore.Mvc.Rendering.ViewContext ViewContext { get; set; }
+    public System.Collections.Generic.Dictionary<System.String, System.Int32> Tags { get; set; }
+     = new System.Collections.Generic.Dictionary<System.String, System.Int32>();
+    public override async global::System.Threading.Tasks.Task ProcessAsync(Microsoft.AspNetCore.Razor.TagHelpers.TagHelperContext context, Microsoft.AspNetCore.Razor.TagHelpers.TagHelperOutput output)
+    {
+        (_helper as global::Microsoft.AspNetCore.Mvc.ViewFeatures.IViewContextAware)?.Contextualize(ViewContext);
+        var content = await _helper.InvokeAsync(""TagCloud"", new { Tags });
+        output.TagName = null;
+        output.Content.SetHtmlContent(content);
+    }
+}
+", vcthClass.Content);
+        }
+
+        private RazorCodeDocument CreateDocument(string content)
+        {
+            using (var stream = new MemoryStream())
+            {
+                var bytes = Encoding.UTF8.GetBytes(content);
+                stream.Write(bytes, 0, bytes.Length);
+                stream.Seek(0L, SeekOrigin.Begin);
+
+                var source = RazorSourceDocument.ReadFrom(stream, "test.cshtml");
+                return RazorCodeDocument.Create(source);
+            }
+        }
+
+        private RazorEngine CreateEngine(params TagHelperDescriptor[] tagHelpers)
+        {
+            return RazorEngine.Create(b =>
+            {
+                b.Features.Add(new MvcViewDocumentClassifierPass());
+
+                b.Features.Add(new TagHelperFeature(tagHelpers));
+            });
+        }
+
+        private DocumentIRNode CreateIRDocument(RazorEngine engine, RazorCodeDocument codeDocument)
+        {
+            for (var i = 0; i < engine.Phases.Count; i++)
+            {
+                var phase = engine.Phases[i];
+                phase.Execute(codeDocument);
+
+                if (phase is IRazorIRPhase)
+                {
+                    break;
+                }
+            }
+
+            return codeDocument.GetIRDocument();
+        }
+
+        private ClassDeclarationIRNode FindClassNode(RazorIRNode node)
+        {
+            var visitor = new ClassDeclarationNodeVisitor();
+            visitor.Visit(node);
+            return visitor.Node;
+        }
+
+        private TagHelperIRNode FindTagHelperNode(RazorIRNode node)
+        {
+            var visitor = new TagHelperNodeVisitor();
+            visitor.Visit(node);
+            return visitor.Node;
+        }
+
+        private string GetCSharpContent(RazorIRNode node)
```

Comment:
Yeah the way I want to do this is through the public API of Razor rather than helpers like this, but that becomes blocking. Good suggesting for improving the IR.